### PR TITLE
Issue636 supplier timeout assertions

### PR DESCRIPTION
**Pull request by nickgrealy**
## Overview

- #636 Introduced `Supplier` variants of the timeout assertions

---

I hereby agree to the terms of the JUnit Contributor License Agreement.

Fyi @jbduncan, @marcphilipp, @sbrannen

### CODE REVIEW

**Comment by nickgrealy** [comment]
PR ready for merge - over to you?

Fyi @jbduncan, @marcphilipp, @sbrannen

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,4 @@ bin/
 *.graphml
 coverage.db*
 clover.license
+local.properties
```
Comment (marcphilipp) [inline .gitignore]: What's that?

Comment (nickgrealy) [inline .gitignore]: @marcphilipp - good catch - the file was created in error (by Android Studio - don't ask). I've removed this line.

**Comment by nickgrealy** [comment]
That's a bit harsh... is there anyway to retrigger without a new commit?

https://travis-ci.org/junit-team/junit5/jobs/199114352#L446

```
FAILURE: Build failed with an exception.
* What went wrong:
Could not resolve all dependencies for configuration ':junit-platform-console:shadowed'.
> Could not resolve net.sf.jopt-simple:jopt-simple:5.0.3.
  Required by:
      project :junit-platform-console
   > Could not resolve net.sf.jopt-simple:jopt-simple:5.0.3.
      > Could not get resource 'https://repo1.maven.org/maven2/net/sf/jopt-simple/jopt-simple/5.0.3/jopt-simple-5.0.3.pom'.
         > Could not HEAD 'https://repo1.maven.org/maven2/net/sf/jopt-simple/jopt-simple/5.0.3/jopt-simple-5.0.3.pom'.
            > handshake alert:  unrecognized_name
```

**Comment by nickgrealy** [comment]
@marcphilipp  - help? https://travis-ci.org/junit-team/junit5/jobs/199119753#L680

**Comment by marcphilipp** [comment]
I've retriggered the Travis build.

**Comment by nickgrealy** [comment]
@marcphilipp  - thanks that did the trick!

**Comment by marcphilipp** [comment]
I am gonna remove the last commit, squash the rest and merge it.

**Comment by marcphilipp** [comment]
Thanks, @nickgrealy! 👍

**Pull request closed without merging**